Fluid/DefaultTemplateCache.cs

```diff
@@ -10,25 +11,28 @@ namespace Fluid;
 /// </summary>
 sealed class TemplateCache : ITemplateCache
 {
-    record struct CachedTemplate(string Name, DateTimeOffset LastModified, IFluidTemplate Template);
+    record struct CachedTemplate(DateTimeOffset LastModified, IFluidTemplate Template);
 
     private readonly ConcurrentDictionary<string, CachedTemplate> _cache;
 
     public TemplateCache()
     {
-        _cache = new(Environment.OSVersion.Platform == PlatformID.Unix ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
+        // Use case-insensitive comparison only on Windows. Create a dedicated cache entry in other cases, even
+        // on MacOS when the file system coulb be case-sensitive too.
+
+        _cache = new(RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? StringComparer.OrdinalIgnoreCase : StringComparer.Ordinal);
     }
 
     public bool TryGetTemplate(IFileInfo fileInfo, out IFluidTemplate template)
     {
         template = null;
 
-        if (_cache.TryGetValue(fileInfo.Name, out var cachedTemplate))
+        if (_cache.TryGetValue(fileInfo.PhysicalPath, out var cachedTemplate))
```

Comment:
Physical path can be null for in-memory filesystems. I think this will break people unless you require this at some other level.

Reply:
This is problematic because `IFileInfo` is very limited here:

```c#
        //
        // Summary:
        //     The name of the file or directory, not including any path.
        string Name { get; }
        //
        // Summary:
        //     The path to the file, including the file name. Return null if the file is not
        //     directly accessible.
        string? PhysicalPath { get; }
```